contracts/EthereumLightClient.sol

```diff
@@ -87,12 +92,36 @@ contract EthereumLightClient is Ethash, Initializable, PausableUpgradeable {
         _setFirstBlock(storedBlock);
     }
 
+    function _deleteBlock(uint256 blockNo) private {
+        uint256[] storage blockHashes = blocksByHeight[blockNo];
+        for(uint256 j = 0; j < blockHashes.length; j++) {
+            delete blocks[blockHashes[j]];
+        }
+        delete blocksByHeight[blockNo];
+        delete blocksByHeightExisting[blockNo];
+        delete finalizedBlocks[blockNo];
+        delete verifiedBlocks[blockNo];
+    }
+ 
+    function _pruneBlocks(uint256 pruneTime) private {
+        uint256 blockCur = oldestBlockStored;
+        for(uint256 i = 0; i < MAX_PRUNE_ONCE; i++) {
+            uint256 blockNo = blockCur++;
+            if(!blocksByHeightExisting[blockNo]) continue;
+            uint256 blockHash = blocksByHeight[blockNo][0];
```

Comment:
I think this will fail to delete uncle blocks at same height.

Reply:
why? `_deleteBlock()` will delete all the blocks of the same height.

Reply:
Oh yep, youre right my bad